tee-worker/omni-executor/native-task-handler/src/lib.rs

```diff
@@ -759,13 +1078,47 @@ pub async fn handle_native_task<
 						},
 					};
 
-				// Check if UserOperation is signed
+				// Check if UserOperation is signed, if not:
```

Comment:
If I construct and sign UserOp by my own that means i can specify `exchange_rate` - but this PR should ensure it's filled by worker. Does it mean we've got a gap to fill?

Reply:
That's a good point.
It can imply a broader problem tho => does it mean that user could use our bundler freely by signing the userOp? (as bundler would just verify the auth and accept it)

Shall we disallow paymaster usage if userOp.signature is non-empty?

Reply:
But I'm not talking about ERC20 paymaster only - but SimplePaymaster that we currently have too.

The user (FE) could construct a userOp with signature and rpc-auth that passes the bundler's check and bundler will submit it to entrypoint (so paymaster would sponsor the gas), right?

Reply:
After discussion, we'll manage it in a simple way:  **use of our paymaster requires an unsiged userOp**.

It means if the user submits a **signed** userOp, we expect that the target paymaster isn't ours (read from config)

Reply:
I updated the logic, see the comment @kziemianek :

```
// Whitelisted paymaster addresses that we support (deployed by heima), the logic is:
// - Unsigned userOp: If paymaster specified, **must** be whitelisted
// - Signed userOp: Only allowed if paymasterAndData is empty (technically we could still relay it, but we are a private bundler and don't want to relay arbitrary userOps)
//
// These addresses are assumed to be the same across all EVM chains
//
// Note: the SimplePaymaster should be gradually deprecated in favor of the ERC20PaymasterV1.
//       Theoretically user could construct an unsiged userOp to use SimplePaymaster "freely"
```